x-pack/plugins/security_solution/public/detection_engine/rule_management/hooks/use_alert_suppression.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback } from 'react';
+import type { Type } from '@kbn/securitysolution-io-ts-alerting-types';
+import { isSuppressibleAlertRule } from '../../../../common/detection_engine/utils';
+import { useIsExperimentalFeatureEnabled } from '../../../common/hooks/use_experimental_features';
+export interface UseAlertSuppressionReturn {
+  isSuppressionEnabled: boolean;
+}
+
+export const useAlertSuppression = (ruleType?: Type): UseAlertSuppressionReturn => {
```

Comment:
Nit: `useAlertSuppressionFeatureFlag` or `useIsAlertSuppressionEnabled`

Reply:
Hook at this point returns `isSuppressionEnabled`, which leaves space for it future enhancements if needed. 
So, I would leave it as `useAlertSuppression` to reflect it.

cc: @WafaaNasr , what's your thoughts on this?

Reply:
The original intention behind creating the `useAlertSuppression` Hook was to consolidate all aspects related to suppression. This includes tasks such as displaying or hiding specific fields based on backend conditions and minimizing duplicated logic within rule creation, editing components, and rule management.